pkg/search/proxy/store/multi_cluster_cache.go

```diff
@@ -146,15 +146,16 @@ func (c *MultiClusterCache) Get(ctx context.Context, gvr schema.GroupVersionReso
 		return nil, err
 	}
 
-	accessor, err := meta.Accessor(obj)
+	cloneObj := obj.DeepCopyObject()
+	accessor, err := meta.Accessor(cloneObj)
 	if err != nil {
 		return nil, err
 	}
 
 	mrv := newMultiClusterResourceVersionWithCapacity(1)
 	mrv.set(clusterName, accessor.GetResourceVersion())
 	accessor.SetResourceVersion(mrv.String())
-	return obj, err
+	return cloneObj, err
```

Comment:
I think you don't need copy it every time when call the get method,  just copy it whenever need to modify the object

Reply:
`accessor.SetResourceVersion(mrv.String())`  it is just modifying it.

Reply:
so, why not deepcopy the object when call the accessor.SetResourceVersion(mrv.String())

Reply:
> so, why not deepcopy the object when call the accessor.SetResourceVersion(mrv.String())

Do you mean as below? I don't think it has too much different. 
```go
	mrv := newMultiClusterResourceVersionWithCapacity(1)
	mrv.set(clusterName, accessor.GetResourceVersion())

	cloneObj := obj.DeepCopyObject()
	accessor, err := meta.Accessor(cloneObj)
	if err != nil {
		return nil, err
	}
	accessor.SetResourceVersion(mrv.String())
	return cloneObj, err
```

Reply:
yeah. I think it will reduce the memory copy and save some memory

Reply:
In fact, err will not occur normally, and copy will be always done